src/components/AnimateSharedLayout/utils/batcher.ts

```diff
@@ -19,24 +22,14 @@ export function createBatcher(): SyncLayoutBatcher {
 
     return {
         add: (child) => queue.add(child),
-        flush: ({ measureLayout, layoutReady, parent } = defaultHandler) => {
+        flush: ({ layoutReady, parent } = defaultHandler) => {
             const order = Array.from(queue).sort(compareByDepth)
 
-            const resetAndMeasure = () => {
-                /**
-                 * Write: Reset any transforms on children elements so we can read their actual layout
-                 */
-                order.forEach((child) => child.resetTransform())
-
-                /**
-                 * Read: Measure the actual layout
-                 */
-                order.forEach(measureLayout)
-            }
-
             parent
-                ? parent.withoutTransform(resetAndMeasure)
-                : resetAndMeasure()
+                ? withoutTreeTransform(parent, () => {
+                      batchResetAndMeasure(order)
+                  })
+                : batchResetAndMeasure(order)
```

Comment:
Why is this a ternary if we don't use the return value?
And even if the return value was used, I think it would be much clearer to just have an if-statement, because of the function calls and callbacks.